bftengine/src/preprocessor/messages/PreProcessResultMsg.cpp

```diff
@@ -105,29 +98,33 @@ std::optional<std::string> PreProcessResultMsg::validatePreProcessResultSignatur
   return {};
 }
 
-std::string PreProcessResultSignature::serializeResultSignatureList(const std::list<PreProcessResultSignature>& sigs) {
+std::string PreProcessResultSignature::serializeResultSignatures(const std::set<PreProcessResultSignature>& signatures,
+                                                                 const uint16_t numOfRequiredSignatures) {
   size_t buf_len = 0;
-  for (const auto& sig : sigs) {
-    buf_len += sizeof(sig.sender_replica) + sizeof(sig.pre_process_result) + sizeof(uint32_t) + sig.signature.size();
+  auto i = 0;
+  for (const auto& s : signatures) {
+    buf_len += sizeof(s.sender_replica) + sizeof(s.pre_process_result) + sizeof(uint32_t) + s.signature.size();
+    if (++i == numOfRequiredSignatures) break;
```

Comment:
I would leave a previous logic serializing all available signatures to avoid edge cases in ReplicaImp signatures verification.

Reply:
previous logic performed std::list::resize(numOfRequiredSignatures) , std::set does not support resize, so I did it like this.
It's essentially the same